examples/src/main/java/io/grpc/examples/dualstack/ExampleDualStackNameResolver.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.dualstack;
+
+import static io.grpc.examples.loadbalance.LoadBalanceClient.exampleServiceName;
+
+import com.google.common.collect.ImmutableMap;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver;
+import io.grpc.Status;
+import io.grpc.examples.loadbalance.LoadBalanceServer;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ExampleDualStackNameResolver extends NameResolver {
+
+    // This is a fake name resolver, so we just hard code the address here.
+    private static final ImmutableMap<String, List<List<SocketAddress>>> addrStore =
+        ImmutableMap.<String, List<List<SocketAddress>>>builder()
+        .put(exampleServiceName,
+            Arrays.stream(LoadBalanceServer.SERVER_PORTS)
+                .mapToObj(port -> getLocalAddrs(port))
+                .collect(Collectors.toList())
+        )
+        .build();
+
+    private Listener2 listener;
+
+    private final URI uri;
+
+    public ExampleDualStackNameResolver(URI targetUri) {
+        this.uri = targetUri;
+    }
+
+    private static List<SocketAddress> getLocalAddrs(int port) {
+        return Arrays.asList(
+            new InetSocketAddress("127.0.0.1", port),
+            new InetSocketAddress("::1", port));
+    }
+
+    @Override
+    public String getServiceAuthority() {
+        // Be consistent with behavior in grpc-go, authority is saved in Host field of URI.
```

Comment:
Oh, that's broken in the current example. It is not the host of the URI in many ways: 1) it should include the port, 2) it will be in the uri's path, 3) the only string ever used in the example doesn't have a host. This would more ordinarily be `uri.getPath().substring(1)`.